src/tests/ftest/slurm_setup.py

```diff
@@ -0,0 +1,281 @@
+
+    # Copy the slurm*example.conf files to /etc/slurm/
+    if execute_cluster_cmds(all_nodes, copy_list, args.sudo) > 0:
+        exit(1)
+
+    cmd_list = [
+        "sed -i -e 's/ControlMachine=linux0/ControlMachine={}/g' {}".format(
+            args.control, slurm_conf),
+        "sed -i -e 's/ClusterName=linux/ClusterName=ci_cluster/g' {}".format(
+            slurm_conf),
+        "sed -i -e 's/SlurmUser=slurm/SlurmUser={}/g' {}".format(
+            args.user, slurm_conf),
+        "sed -i -e 's/NodeName/#NodeName/g' {}".format(
+            slurm_conf),
+        ]
+
+    # This info needs to be gathered from every node that can run a slurm job
+    command = r"lscpu | grep -E '(Socket|Core|Thread)\(s\)'"
+    task = run_task(all_nodes, command)
+    for output, nodes in task.iter_buffers():
+        info = {
+            data[0]: data[1]
+            for data in re.findall(
+                r"(Socket|Core|Thread).*:\s+(\d+)", str(output))
+            if len(data) > 1}
+
+        if "Socket" not in info or "Core" not in info or "Thread" not in info:
+            # Did not find value for socket|core|thread so do not
+            # include in config file
+            pass
+        cmd_list.append("echo \"NodeName={0} Sockets={1} CoresPerSocket={2} "
+                        "ThreadsPerCore={3}\" >> {4}".format(
+                            NodeSet.fromlist(nodes), info["Socket"],
+                            info["Core"], info["Thread"], slurm_conf))
+
+    #
+    cmd_list.append("echo \"PartitionName=daos_client Nodes={} Default=YES "
+                    "MaxTime=INFINITE State=UP\" >> {}".format(
+                        args.nodes, slurm_conf))
+
+    if execute_cluster_cmds(all_nodes, cmd_list, args.sudo) > 0:
+        exit(1)
+
+
+def execute_cluster_cmds(nodes, cmdlist, sudo=False, timeout=None):
+    """Execute the list of cmds on hostlist nodes.
+
+    Args:
+        nodes (list):  list of nodes
+        cmdlist ([type]): list of cmdlines to execute
+        sudo (str, optional): Execute cmd with sudo privs. Defaults to false.
+
+     Returns:
+        ret_code: returns error code if pcmd fails;
+
+    """
+    for cmd in cmdlist:
+        if sudo:
+            cmd = "sudo {}".format(cmd)
+        result = pcmd(nodes, cmd, True, None, 0)
+        # if at least one node failed or all nodes failed
+        # return on first failure
+        if len(result) > 1 or 0 not in result:
+            return 1
+    return 0
+
+
+def configuring_packages(args, action):
+    """Install required slurm and munge packages.
+
+    Args:
+        args (Namespace): Commandline arguments
+        action (str):  install or remove
+
+    """
+    # Install yum packages on control and compute nodes
+    all_nodes = NodeSet("{},{}".format(str(args.control), str(args.nodes)))
+    cmd_list = []
+    for package in package_list:
+        logging.info("{} {} on {}".format(action, package, all_nodes))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments